src/cpp/src/llm_pipeline.cpp

```diff
@@ -119,14 +126,36 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
                 auto new_templated_chat_history  = m_tokenizer.apply_chat_template(m_history, add_generation_prompt);
                 // Do not add special tokens in chat scenario to be aligned with HF.
                 auto new_chat_tokens = m_tokenizer.encode(new_templated_chat_history, ov::genai::add_special_tokens(false));
-                if (m_is_cache_empty) {
-                    encoded_input = new_chat_tokens;
-                } else {
-                    auto prev_chat_tokens = m_tokenizer.encode(m_templated_chat_history, ov::genai::add_special_tokens(false));
+                auto prev_chat_tokens = m_tokenizer.encode(m_templated_chat_history, ov::genai::add_special_tokens(false));
+
+                // some symbols combinations can be encoded by the tokenizer in different ways
+                // if we met sequence with such combination of symbols, we cannot correctly subtract the new history from the old history
+                // and find the difference as a prompt, so let's check it out and use the whole history in this case
+                if (!m_tokenized_chat_history.empty()) {
+                    auto stop_tokens = config.stop_token_ids;
+                    // config could be reset by user and stop_tokens could be empty
+                    // but model/tokenizer still will rely to eos token, so let's add it
+                    stop_tokens.insert(m_tokenizer.get_eos_token_id());
```

Comment:
I think we need to add the following block of code:
```c++
        // If eos_token_id was not provided, take value from default m_generation_config
        if (config.eos_token_id == -1)
            config.set_eos_token_id(m_generation_config.eos_token_id);
        config.validate();
```
as we have in other places.

Reply:
replaced here https://github.com/openvinotoolkit/openvino.genai/pull/1347